File: java/src/main/java/toolgood/flowVision/Flows/AppInitValueWork.java
package toolgood.flowVision.Flows;

import com.alibaba.fastjson2.JSONArray;
import com.alibaba.fastjson2.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class AppInitValueWork {
    public String Name;
    public toolgood.flowVision.Flows.Enums.InputType InputType;
    public List<SettingFormulaItemWork> Conditions;

    public boolean IsThrowError;
    public String ErrorMessage;

    final static AppInitValueWork parse(JSONObject jsonObject) {
        AppInitValueWork result = new AppInitValueWork();
        result.Name = jsonObject.getString("name");
        result.InputType = toolgood.flowVision.Flows.Enums.InputType.intToEnum(jsonObject.getIntValue("inputType"));
        result.Conditions = new ArrayList<>();
        JSONArray conditions = jsonObject.getJSONArray("conditions");
        for (Object item : conditions) {
            if (item instanceof JSONObject) {
                JSONObject jsonObject1 = (JSONObject) item;
                SettingFormulaItemWork itemWork = SettingFormulaItemWork.parse(jsonObject1);
                if (itemWork != null) {
                    result.Conditions.add(itemWork);
                }
            }
        }
        return result;
    }

    public void Init(ProjectWork work) {
        for (int i = 0; i < Conditions.size(); i++) {
            SettingFormulaItemWork item = Conditions.get(i);
            item.Init(work);
        }
    }
}

File: editor/ToolGood.FlowWork.Applications/Flows/AppInitValueWork.cs
using System.Text.Json.Serialization;
using ToolGood.FlowWork.Applications.Engines;

namespace ToolGood.FlowWork.Flows
{
	public sealed class AppInitValueWork
	{
		/// <summary>
		/// 名称
		/// </summary>
		public string Name { get; set; }

		/// <summary>
		/// 输入类型
		/// </summary>
		public InputType InputType { get; set; }// 输入类型

		[System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
		public List<SettingFormulaItemWork> Conditions { get; set; }

		/// <summary>
		/// 抛出错误
		/// </summary>
		public bool IsThrowError { get; set; }

		/// <summary>
		/// 错误信息
		/// </summary>
		[System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
		public string ErrorMessage { get; set; }

		internal void Init(ProjectWork work)
		{
			for (int i = 0; i < Conditions.Count; i++) {
				var item = Conditions[i];
				item.Init(work);
			}
		}

		internal string GetMatchFormula(FlowEngine engine)
		{
			for (int i = 0; i < Conditions.Count; i++) {
				var item = Conditions[i];
				if (item.Check(engine)) {
					return item.Formula;
				}
			}
			return null;
		}
	}
}